finality-aleph/src/party.rs

```diff
@@ -1,101 +1,88 @@
     SC: SelectChain<B> + 'static,
 {
-    pub(crate) fn new(
-        client: Arc<C>,
-        network: N,
-        select_chain: SC,
-        auth_keystore: AuthorityKeystore,
-    ) -> Self {
-        let network = ConsensusNetwork::new(network, "/cardinals/aleph/1");
-
-        ConsensusParty {
-            network,
-            client,
-            select_chain,
-            auth_keystore,
-            _phantom: PhantomData,
-        }
-    }
+    // TODO now it runs just a single instance of consensus but later it will
+    // orchestrate managing multiple instances for differents epoch
+    let AlephParams {
+        config:
+            crate::AlephConfig {
+                network,
+                consensus_config,
+                client,
+                select_chain,
+                spawn_handle,
+                auth_keystore,
+                authorities,
+            },
+    } = aleph_params;
+    let network = ConsensusNetwork::new(network, "/cardinals/aleph/1");
+    let spawn_handle: SpawnHandle = spawn_handle.into();
 
-    pub(crate) async fn run(
-        &self,
-        authorities: Vec<AuthorityId>,
-        conf: ConsensusConfig<NodeId>,
-        spawn_handle: SpawnHandle,
-    ) {
-        // TODO now it runs just a single instance of consensus but later it will
-        // orchestrate managing multiple instances for differents epoch
+    let (net_command_tx, net_command_rx) = mpsc::unbounded();
+    let network = network.clone();
+    let task = {
+        let network = network.clone();
+        async move { network.run(net_command_rx).await }
+    };
 
-        let (net_command_tx, net_command_rx) = mpsc::unbounded();
-        let network = self.network.clone();
-        let task = async move { network.run(net_command_rx).await };
+    spawn_handle.0.spawn("aleph/network", task);
 
-        spawn_handle.0.spawn("aleph/network", task);
+    let epoch_id = EpochId(0);
+    let network_event_rx = network.start_epoch(epoch_id, authorities.clone());
+    let hashing = Blake2Hasher::hash;
 
-        let epoch_id = EpochId(0);
-        let rx_network = self.network.start_epoch(epoch_id, authorities.clone());
-        let hashing = Blake2Hasher::hash;
-        let auth_keystore = self.auth_keystore.clone();
-        let client = self.client.clone();
-        let select_chain = self.select_chain.clone();
-        let mut env: Environment<B, H256, C, BE, SC> = Environment::new(
-            client,
-            select_chain,
-            net_command_tx,
-            rx_network,
-            None,
-            authorities,
-            auth_keystore,
-            hashing,
-            epoch_id,
-        );
+    let (notification_in_tx, notification_in_rx) = mpsc::unbounded();
+    let (notification_out_tx, notification_out_rx) = mpsc::unbounded();
+    // Making `rush::Consensus` accept only tokio sender sinks seems a questionable choice...
+    let (order_tx, order_rx) = tokio::sync::mpsc::unbounded_channel();
```

Comment:
A lot of txes and rxes are created here, but anyway, it seems to be the most reasonable approach to link different components with channels. Alternatively we could return sinks (txes) and streams (rxes) in the constructors of the components, but this would require forwarding the streams to sinks, which generates many futures to poll. Yet another solutions would be to choose a convention that e.g. the constructors create channels, keep the sinks return streams, keep the sinks, and the streams are initialized after the construction, which is not a clean solution neither.

Reply:
Yeah, I see what you are saying. I would keep it as it is -- I think this will be the only place with mess like that.